app/question.js

```diff
@@ -63,45 +73,74 @@ define(['jquery', 'app/messages', 'app/config', 'app/string'], function ($, mess
   }
 
   // Handle how to move to the next question
-  // depending on the tutorial status.
+  // depending on the tutorial status and algorithm.
   function nextQuestion() {
-    if (inTutorial) {
-      currentItemIndex++;
-      inTutorial = checkTutorialStatus();
-    } else if (answerWasCorrect) {
-      items.splice(currentItemIndex, 1);
-      currentItemIndex %= items.length;
-    } else {
-      currentItemIndex = (currentItemIndex + 1) % items.length;
+
+    switch(config.constant("ALGORITHM")) {
+      case "flashcard":
+        // Use flaschcard method to determine next question
+        if (inTutorial) {
+          currentItemIndex++;
+          inTutorial = checkTutorialStatus();
+        } else if (answerWasCorrect) {
+          items.splice(currentItemIndex, 1);
+          currentItemIndex %= items.length;
+        } else {
+          currentItemIndex = (currentItemIndex + 1) % items.length;
+        }
+        break;
+      case "slimstampen":
+        // Update the response list in order to determine next question
+        newResponse = {
+          factId: items[currentItemIndex].id,
+          timeCreated: timeCreated,
+          number: 0,
+          data : JSON.stringify({ reactionTime: firstKeyPress,sessionTime: 0, correct: answerWasCorrect })
+        };
+        responseList.push(newResponse);
+        // Use slimstampen method to determine next question
+        var newQuestion = slimstampen.getNextFact(firstKeyPress, items, responseList);
+        currentItemIndex = items.indexOf(newQuestion);
+        break;
     }
+    timeCreated = measureTime(startTime);
+    firstKeyPress = 0;
   }
 
   function showTutorialInstruction() {
-    $("#question").append("<br><b>Type the answer:</b> " + items[currentItemIndex].item_answer);
+    $("#question").append("<br><b>Type the answer:</b> " + items[currentItemIndex].answer);
```

Comment:
Since we will have internationalization module, these messages will be moved to language files